OsmAnd/src/net/osmand/plus/myplaces/AvailableGPXFragment.java

```diff
@@ -732,18 +732,55 @@ public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
 			@Override
 			public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
 				if (selectedItems.isEmpty()) {
-					Toast.makeText(getActivity(),
-							app.getString(R.string.local_index_no_items_to_do, actionButton.toLowerCase()),
-							Toast.LENGTH_SHORT).show();
+					app.showShortToastMessage(app.getString(R.string.local_index_no_items_to_do, actionButton.toLowerCase()));
 					return true;
 				}
+				if (actionResId == R.string.shared_string_delete){
+					AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+					builder.setMessage(getString(R.string.local_index_action_do, actionButton.toLowerCase(),
+							String.valueOf(selectedItems.size())));
+					builder.setPositiveButton(actionButton, listener);
+					builder.setNegativeButton(R.string.shared_string_cancel, null);
+					builder.show();
+				} else if (actionResId == R.string.local_index_mi_upload_gpx){
+					List<SelectableItem> allTracks = new ArrayList<>();
+					List<SelectableItem> selectedTracks = new ArrayList<>();
+					for (GpxInfo gpxInfo: selectedItems) {
+						SelectableItem s = createSelectableItem(gpxInfo);
+						s.setObject(gpxInfo);
+						s.setIconId(R.drawable.ic_notification_track);
+						s.setTitle(gpxInfo.name);
+						allTracks.add(s);
+						selectedTracks.add(s);
+					}
+					ExportGPXMultipleSelectionBottomSheet dialog = ExportGPXMultipleSelectionBottomSheet.showInstance(getMyActivity(), allTracks, selectedTracks);
+					dialog.setDialogStateListener(new SelectionBottomSheet.DialogStateListener() {
+						@Override
+						public void onDialogCreated() {dialog.setTitle(actionButton);
+							dialog.setApplyButtonTitle(app.getString(R.string.shared_string_continue));
+							String total = app.getString(R.string.shared_string_total);
+							long size = 0;
+							for (SelectableItem s: allTracks) {
+								size += ((GpxInfo) s.getObject()).getSize();
+							}
+							dialog.setTitleDescription(app.getString(R.string.ltr_or_rtl_combine_via_colon, total, AndroidUtils.formatSize(app, size)));
+						}
 
-				AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
-				builder.setMessage(getString(R.string.local_index_action_do, actionButton.toLowerCase(),
-						String.valueOf(selectedItems.size())));
-				builder.setPositiveButton(actionButton, listener);
-				builder.setNegativeButton(R.string.shared_string_cancel, null);
-				builder.show();
+						@Override
+						public void onCloseDialog() {
+						}
+					});
+					dialog.setOnApplySelectionListener(selItems -> {
+						GpxInfo[] gpxInfo = new GpxInfo[selItems.size()];
```

Comment:
Probably you can change and use listener which is passed to openSelectionMode and send files here - 
https://github.com/osmandapp/Osmand/blob/6d2ef9da0af625e161cdfe9aa32f266920bf05e3/OsmAnd/src/net/osmand/plus/osmedit/OsmEditingPlugin.java#L414